### PR TITLE
Trigger docker pull for new image

### PR DESCRIPTION
**Pull request by CallumMcGregorAD**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@de0132cf8cdedb79975c6d42b77eb7ea193cf28e
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: Think this can be `aws-actions/amazon-ecs-deploy-task-definition@v1`

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Build image
+        id: build-image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
```
Comment (bethcryer) [inline .github/workflows/deploy.yml:62]: Add the `Download task definition` stage I sent to you on slack

Comment (bethcryer) [inline .github/workflows/deploy.yml:62]: We probably also want to include the `Fill in the new image ID in the Amazon ECS task definition` stage from the github documentation example of this: [link](https://docs.github.com/en/actions/deployment/deploying-to-your-cloud-provider/deploying-to-amazon-elastic-container-service)  

We can then use the image ID to rollback to a previous image in the ECS if we need to

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+on:
+  push:
+    branches: "release-*"
+
+env:
+    AWS_REGION: eu-west-2
+    ECS_SERVICE: answerking-dotnet-api-ecs-service
+    ECS_CLUSTER: answerking-dotnet-api-ecs-cluster
+    ECS_TASK_DEFINITION: .aws/task-definition.json
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: This can just be `task-definition.json` now

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    name: Build and publish a docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Log in to github registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
```
Comment (bethcryer) [inline .github/workflows/deploy.yml:36]: You need to copy the environment variables from `docker-build.yml` into this action's `env:` section

Comment (bethcryer) [reply .github/workflows/deploy.yml:36]: Should also delete` docker-build.yml` if we're replacing it with this

**Review by bethcryer — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        with:
+          context: .
+          file: ./dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Download task definition
+          run: |
+            aws ecs describe-task-definition --task-definition answerking-dotnet-api-task --query taskDefinition > task-definition.json
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: Can swap out `task-definition.json` for `${{ env.ECS_TASK_DEFINITION }}`

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+          
+    name: Build and publish a docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: Remove this bit I've highlighted - this is the job definition section copied from the old docker yml, but in this yml we already define the job earlier and ran the checkout action.

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          run: |
+            aws ecs describe-task-definition --task-definition answerking-dotnet-api-task --query taskDefinition > task-definition.json
+            
+      - name: Fill in the new image ID in the Amazon ECS task definition
+        id: task-def
+        uses: aws-actions/amazon-ecs-render-task-definition@v1
+        with:
+          task-definition: ${{ env.ECS_TASK_DEFINITION }}
+          container-name: ${{ env.CONTAINER_NAME }}
+          image: ${{ steps.build-image.outputs.image }}
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: This needs to be `steps.build-image.outputs.imageid`

**Review by bethcryer — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: production
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@13d241b293754004c80624b5567555c4a39ffbe3
```
Comment (bethcryer) [inline .github/workflows/deploy.yml]: Missed this, should also be @v1

Comment (CallumMcGregorAD) [reply .github/workflows/deploy.yml]: Thought I'd got them all, changed now

**Review by bethcryer — DISMISSED** [review]

**Review by DWatson1993 — DISMISSED** [review]

**Review by bethcryer — APPROVED** [review]

**Review by DWatson1993 — APPROVED** [review]

**Pull request merged**